client/src/main/java/org/eclipse/hono/client/Command.java

```diff
@@ -242,6 +243,15 @@ public String getDeviceId() {
         return deviceId;
     }
 
+    /**
+     * Sets the identifier of the gateway this command is to be sent to.
+     *
+     * @param gatewayId The gateway identifier.
+     */
+    public void setGatewayId(final String gatewayId) {
+        this.deviceId = Optional.ofNullable(gatewayId).orElseGet(this::getOriginalDeviceId);
```

Comment:
I've adapted the javadoc here.

> BTW is it really expected/intended that this method is being invoked with `null` as the argument?

I've mainly added that to comply with the corresponding new `Command` interface method. Since `getGatewayId` may return `null` there, I think it's ok to allow resetting a command to such a state.